### PR TITLE
Upgrade to spring boot 3 with Camunda 7.20

### PR DESCRIPTION
**Pull request by adusumillipraveen**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DTSPO-15649 

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,7 +12,7 @@ plugins {
     id 'checkstyle'
     id 'io.spring.dependency-management' version '1.1.4'
     id 'com.github.ben-manes.versions' version '0.50.0'
-    id 'org.springframework.boot' version '2.7.18'
+    id 'org.springframework.boot' version '3.1.5'
```
Comment (adusumillipraveen) [inline build.gradle:15]: Cannot do 3.2 as https://docs.camunda.org/manual/7.20/user-guide/spring-boot-integration/version-compatibility/

File: build.gradle [hunk 2]
```diff
@@ -163,22 +156,24 @@ dependencies {
...
+    implementation group: 'org.camunda.bpm', name: 'camunda-engine-spring-6'
 
     implementation 'org.springframework.boot:spring-boot-starter-oauth2-client'
     implementation 'org.springframework.boot:spring-boot-starter-security'
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.springframework.boot:spring-boot-starter-jdbc'
     implementation 'org.springframework.security.oauth:spring-security-oauth2:2.5.2.RELEASE'
     implementation 'org.springframework.boot:spring-boot-starter-actuator'
-
+    implementation group: 'jakarta.servlet', name: 'jakarta.servlet-api'
```
Comment (adusumillipraveen) [inline build.gradle:167]: Needed with the Javax to Jakarta change in spring 3

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java [hunk 3]
```diff
@@ -6,21 +6,30 @@
...
+import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
 
 import java.util.Collections;
 
 @Configuration
 @ConditionalOnProperty(prefix = "camunda.ui.auth", name = "enabled", matchIfMissing = false)
 @EnableWebSecurity
 @Order(100)
-public class WebSecurityWebAppConfig extends WebSecurityConfigurerAdapter {
+public class WebSecurityWebAppConfig {
```
Comment (adusumillipraveen) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java:26]: See https://spring.io/security/cve-2023-34035

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -4,12 +4,18 @@ logging:
 management:
   endpoint:
     health:
+      probes:
+        enabled: true
```
Comment (adusumillipraveen) [inline src/main/resources/application.yaml:8]: Some cannot get readiness/livness probes to work without these

**Review by timja — APPROVED** [review]
nice work

**Pull request closed without merging**